src/ArrowTypes/Project.toml

```diff
@@ -18,11 +18,17 @@
 name = "ArrowTypes"
 uuid = "31f734f8-188a-4ce0-8406-c8a06bd891cd"
 authors = ["quinnj <quinn.jacobd@gmail.com>"]
-version = "2.3.0"
+version = "2.4.0"
 
 [deps]
+
+[weakdeps]
 Sockets = "6462fe0b-24de-5631-8697-dd941f90decc"
 UUIDs = "cf7118a7-6976-5b1a-9a39-7adc72f591a4"
 
+[extensions]
+ArrowTypesSocketsExt = "Sockets"
+ArrowTypesUUIDsExt = "UUIDs"
+
 [compat]
-julia = "1.0"
+julia = "1.9"
```

Comment:
Why not keep the existing Julia version compat and do this instead? https://pkgdocs.julialang.org/v1/creating-packages/#Transition-from-normal-dependency-to-extension